### DISCUSSION TITLE
init-Event not fired for React-version of Embla

### DISCUSSION THREAD

**Discussion by gruberro** [davidjerleke/embla-carousel#379] [Help] [answered]
Hi all!

I've a question regarding the `init`-event when using Embla with React: the `init`-event is not fired (or at least hookable by the application) once Embla is "ready". This seems to occur starting with version `7`.

I created a tiny demo https://codesandbox.io/s/embla-carousel-progress-react-forked-1f54rg?file=/src/js/EmblaCarousel.js.

So my question is: is this intended behavior, or am I doing something wrong here 😄?

**Comment by davidjerleke** [comment, marked as answer]
Hi @gruberro,

Thank you for your question. That's correct, it's not fired because the `emblaApi` here is a react state variable:

```ts
const [emblaApi] = useEmblaCarousel();
```

By the time Embla is initialised and the `emblaApi` is set, the Embla `init` event has already fired. But you can do the init in an equivalent way with a `useEffect()`:
```ts
useEffect(() => {
  if (!emblaApi) return;

  // here you can run your init code
}, [emblaApi])
```

Note that all other native Embla events except `init` will work as expected though. I hope this explanation makes sense.

Best,
David

**Reply by gruberro** [reply]
Thank you very much for your immediate reply. That helps a lot 👍
